raiden_contracts/tests/test_monitoring_service.py

```diff
@@ -268,6 +268,34 @@ def test_monitor(
     )
 
 
+def test_monitor_on_wrong_token_network_registry(
+    token_network_in_another_token_network_registry: Contract,
+    monitoring_service_external: Contract,
+    monitor_data: Dict,
+    ms_address: HexAddress,
+    web3: Web3,
+) -> None:
+    A, B = monitor_data["participants"]
+
+    # wait until MS is allowed to monitor
+    token_network_in_another_token_network_registry.web3.testing.mine(
+        monitor_data["first_allowed"] - web3.eth.blockNumber
+    )
+
+    # monitor() call fails because the TokenNetwork is not registered on the
+    # supposed TokenNetworkRegistry
+    with pytest.raises(TransactionFailed):
```

Comment:
Now that we have messages on our requires, we could start checking that the tests fail for the right reason. Since we don't measure the coverage of our solidity code, we could have a wrong require and not notice it in any way if our transaction can also fail for a different reason.

Reply:
I agree.  I'll figure out how to check the error messages generated by `require()` messages.

Reply:
This has been done.  https://github.com/raiden-network/raiden-contracts/issues/1158 keeps track of doing the same for all other available `require()` messages.